server/routes/api/user.ts

```diff
@@ -132,31 +155,45 @@ function postApplicationBranchHandler(anonymous: boolean): (request: express.Req
 					};
 				}
 				else {
-					errored = true;
-					response.status(400).json({
-						"error": `"${question.label}" is a required field`
-					});
-					return null;
+					return reportError(`"${question.label}" is a required field`);
 				}
 			}
-			if ((question.type === "select" || question.type === "radio") && Array.isArray(request.body[question.name]) && question.hasOther) {
+			if (question.minCharacterCount && getQuestion("").length < question.minCharacterCount) {
+				return reportError(`Your response to "${question.label}" must have at least ${question.minCharacterCount} characters`);
+			}
+			if (question.maxCharacterCount && getQuestion("").length > question.maxCharacterCount) {
+				return reportError(`Your response to "${question.label}" cannot exceed ${question.maxCharacterCount} characters`);
+			}
+			let wordCount = getQuestion("").trim().split(/\s+/).length;
+			if (getQuestion("").trim().length === 0) {
```

Comment:
getQuestion("") is not truthy @petschekr. This returns undefined for file upload fields and which crashes registration